Source/WebKit/UIProcess/Cocoa/ProcessAssertionCocoa.mm

```diff
@@ -455,6 +462,25 @@ - (void)assertion:(RBSAssertion *)assertion didInvalidateWithError:(NSError *)er
     return !m_wasInvalidated;
 }
 
+ProcessAndUIAssertion::ProcessAndUIAssertion(pid_t pid, const String& reason, ProcessAssertionType assertionType)
+    : ProcessAssertion(pid, reason, assertionType)
+{
+#if PLATFORM(IOS_FAMILY)
+    updateRunInBackgroundCount();
+#endif
+}
+
+#if PLATFORM(IOS_FAMILY)
```

Comment:
nit: I would have just guarded the destructor implementation to make the code a bit more concise.